compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/PsiUtils.kt

```diff
@@ -428,6 +428,14 @@ public fun PsiElement.requireFqName(
       }
     }
 
+  // Check if it's an inner class of this file. A common case is return types of functions for
+  // inner classes, which require no import. Walk up the inner class hierarchy.
+  parents
```

Comment:
This looked very familiar to me, because of `declarationsWithinScope` which is defined just below this function.  It _should have_ caught this edge case, but I missed an edge case.

The problem is that the existing function uses `it.children.filterIsInstance<KtNamedDeclaration>() + it` instead of `it.collectDescendantsOfType<KtNamedDeclaration>() + it`.

If `declarationsWithinScope()` is updated, then this addition shouldn't be necessary.  On the other hand, if this code is here, then the logic which calls `declarationsWithinScope()` isn't necessary.

Reply:
Oh, I wasn't aware. I'm in favor of keeping the new code. I'm a little unsure why you added the extension on `PsiElement`. Because of that resolving the FqName seems to involve more operations. `module.resolveFqNameOrNull(it)` is one of heavier functions.

Since we only care about classes or objects, I think it's safe to filter for that. I updated the code and used `collectDescendantsOfType`. That'a nice utility.